django/db/models/query.py

```diff
@@ -1874,6 +1881,15 @@ def _clone(self):
     def _fetch_all(self):
         if self._result_cache is None:
             self._result_cache = list(self._iterable_class(self))
+            # If we're fetching objects tell them about each other so they can
+            # prefetch for each other during attribute access.
+            if (
```

Comment:
you can save non trivial memory in some edgecasey workloads by not doing this when len of result cache is 1, of course then you need to deal with peers being missing in some other places
this was done on the upstream recently (at my suggestion) https://github.com/tolomea/django-auto-prefetch/commit/a001f140473c6c9c3e8037936671a2c794d12bf6

Reply:
actually you need to handle it not always being there anyway because the __getstate__ is removing it, so it won't be there after deserialization from pickle, currently that doesn't seem to be happening and I think that's probably a bug
you could instead set it to [], however that still means releasing this would break any existing pickled objects